internal/filesystem/filesystem_test.go

```diff
@@ -0,0 +1,185 @@
+package filesystem_test
+
+import (
+	"context"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/djdv/go-filesystem-utils/internal/filesystem"
+)
+
+type (
+	openFileFSMock struct{ fs.FS }
+	streamDirMock  struct {
+		fs.ReadDirFile
+		entries []filesystem.StreamDirEntry
+	}
+)
+
+var (
+	_ filesystem.OpenFileFS    = (*openFileFSMock)(nil)
+	_ filesystem.StreamDirFile = (*streamDirMock)(nil)
+)
+
+func (of *openFileFSMock) OpenFile(name string, _ int, _ fs.FileMode) (fs.File, error) {
+	// NOTE: Mock discards arguments.
+	// We're only interested in seeing the test coverage trace.
+	// The wrapper should follow the [filesystem.OpenFileFS]
+	// type-assertion branch, when passed our mock.
+	return of.FS.Open(name)
+}
+
+func (sd *streamDirMock) StreamDir(ctx context.Context) <-chan filesystem.StreamDirEntry {
+	entries := make(chan filesystem.StreamDirEntry)
+	go func() {
+		defer close(entries)
+		for _, entry := range sd.entries {
+			select {
+			case entries <- entry:
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+	return entries
+}
+
+func TestFilesystem(t *testing.T) {
+	t.Parallel()
+	t.Run("OpenFileFS", openFileFS)
+	t.Run("StreamDir", streamDir)
```

Comment:
These tests should be switched over to use `testing/fstest.MapFS` instead of the host file system.

Reply:
Done in: a81740b63177f48d9e704037f8c1a0dd0567515a